### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by VSeray**
- Installed CanCanCan in the project.
   - Added a role column to the user's table, using migration for this.
   - Ensured a user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.

- Implemented the post deleting functionality. Added the "Delete" button to the view and make sure that only authorized users can see it.

    - A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.

- Implemented the comment deleting functionality. Added the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = current_user
```
Comment (omar-labana) [inline app/controllers/posts_controller.rb:5]: - [ ] According to the project requirements, you should use **CanCanCan** for Post authorization in `Post Controller` bt loading `load_and_authorize_resource`.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @VSeray and @gtekle, 

Excellent work so far working with CanCan! 👌
You have implemented almost all the required features without any bugs or errors. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (VSeray) [reply app/controllers/posts_controller.rb:5]: Hello @omar-labana, thank you for the review, but can you please explain more briefly what I should be implementing and what the issue is?

**Review by dasileker — APPROVED** [review]
### **Hi @VSeray,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (omar-labana) [reply app/controllers/posts_controller.rb:5]: @VSeray Sorry for not replying in time, I see that you have added the missing guard, good luck in the next milestones

**Pull request merged**